src/main/java/org/jenkinsci/plugins/codesonar/models/CodeSonarBuildActionDTO.java

```diff
@@ -77,10 +71,20 @@ protected Object readResolve() {
         
         if (analysisActiveWarnings == null) {
             LOGGER.log(Level.WARNING, "Found empty analysisActiveWarnings on persisted analysis");
+        } else {
```

Comment:
Supposing to load an old Jenkins' build, with a CodeSonarBuildActionDTO which did not yet include the analysisId field, the purpose of this code is to populate that field by retrieving the analysisId from within the original `Analysis` object (`analysisActiveWarnings` specifically).
I need to double check the scenario when you reserialize the DTO object back, after loading its content from an old build descriptor. In such a case we do want that the descriptor will not contain those giant `Analysis` objects, right?

Reply:
> In such a case we do want that the descriptor will not contain those giant Analysis objects, right?

This is a good question.  I am not sure of the correct solution.  I am inclined to think that we should _not_ reserialize the data - at least not in this version of the plugin.  Although I think it would be preferrable to clean up the old data, if someone were to upgrade the plugin and then downgrade later, they would not be able to use the data from their old builds because it would be erased.  There is some risk that the user will want to downgrade, if for example, the new plugin version is not compatible with their old hub, or if there are bugs in the new plugin version that can only be worked-around by downgrading.

Reply:
I believe that at the moment the plugin behaves in that way, so by preserving the old build descriptor.
This would mean that old descriptors will stay as-is, occupying potentially a high amount of disk space, as long as some Jenkins' cleanup process won't delete them, only newly run pipelines will benefit from our improvements.
I think that this would be reasonable.